include/envoy/upstream/retry.h

```diff
@@ -18,18 +18,34 @@ class RetryPriority {
 public:
   virtual ~RetryPriority() = default;
 
+  /**
+   * Function that maps a HostDescription to it's effective priority level in a cluster.
+   * For most cluster types, the mapping is simply `return host.priority()`, but some
+   * cluster types require more complex mapping.
+   */
+  using PriorityMappingFunc =
```

Comment:
probably want to document why the return value is optional here, ie what does it mean to get `absl::nullopt` back